blocks/extensions/plot.py

```diff
@@ -80,52 +79,76 @@ def __init__(self, document, channels, open_browser=False,
         if server_url is None:
             server_url = config.bokeh_server
 
-        self.plots = {}
+        self.data_sources = {}
         self.start_server = start_server
         self.document = document
         self.server_url = server_url
+
+        kwargs.setdefault('after_epoch', True)
+        kwargs.setdefault("before_first_epoch", True)
+
+        # set x_axis_label to the most frequent option
```

Comment:
This is not very elegant, but I didn't want to add parameters to the constructor. Basically I try to infer if the x-axis should be over iterations or batches, instead of just assuming that iteration is ok.